### PR TITLE
[Samples][42.scaleout] Add ScaleoutBot class

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        dialog = withDialog;
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        // Create the storage key for this conversation.
+        String key = String.format(
+            "%s/conversations/%s",
+            turnContext.getActivity().getChannelId(),
+            turnContext.getActivity().getConversation().getId());
```
Comment (fran893) [inline samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java:44]: In C# has the `?` which means that there is a chance that it is null, do we have to add a verification?

File: samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        store = withStore;
+
+        if (withDialog == null) {
+            throw new IllegalArgumentException("withDialog can't be null");
+        }
+        dialog = withDialog;
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
```
Comment (fran893) [inline samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java:39]: Can you add the [javadocs](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/42.scaleout/Bots/ScaleoutBot.cs#L35)?

File: samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                                .thenCompose(success -> {
+                                    // Following a successful save, send any outbound Activities,
+                                    // otherwise retry everything.
+                                    if (success) {
+                                        if (!pairNewState.getLeft().isEmpty()) {
+                                            // This is an actual send on the TurnContext we were given
+                                            // and so will actual do a send this time.
+                                            return turnContext.sendActivities(pairNewState.getLeft())
+                                                .thenApply(result -> null);
+                                        }
```
Comment (fran893) [inline samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java:66]: You are missing the [break](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/42.scaleout/Bots/ScaleoutBot.cs#L69).

**Review by fran893 — CHANGES_REQUESTED** [review]

File: samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java (hunk 2 above)
Comment (Batta32) [reply samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java:48]: Yes, I added as a draft as I had some missing changes that I could apply right now

File: samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java (hunk 3 above)
Comment (Batta32) [reply samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java:80]: Yes, I added as a draft as I had some missing changes that I could apply right now

**Review by fran893 — APPROVED** [review]

**Pull request merged**